atlasdb-config/src/main/java/com/palantir/atlasdb/http/TextDelegateDecoder.java

```diff
@@ -59,15 +57,13 @@ public TextDelegateDecoder(Decoder delegate) {
     }
 
     @Override
-    public Object decode(Response response, Type type) throws IOException, DecodeException, FeignException {
-        Collection<String> contentTypes = response.headers().get(HttpHeaders.CONTENT_TYPE);
-        // In the case of multiple content types, or an unknown content type, we'll use the delegate instead.
-        if (contentTypes != null
-                && contentTypes.size() == 1
-                && Iterables.getOnlyElement(contentTypes, "").equals(MediaType.TEXT_PLAIN)) {
+    public Object decode(Response response, Type type) throws IOException, FeignException {
+        if (HeaderAccessUtils.shortcircuitingCaseInsensitiveContainsEntry(
+                response.headers(),
+                HttpHeaders.CONTENT_TYPE,
```

Comment:
(For best overall general performance uppercase would make sense, but given our distribution actually lowercase is best, since most of the time everything is lowercase)